prboom2/src/s_sound.c

```diff
@@ -324,7 +324,7 @@ void S_StartSoundAtVolume(void *origin_p, int sfx_id, int volume)
   // kill old sound
   for (cnum=0 ; cnum<numChannels ; cnum++)
     if (channels[cnum].sfxinfo && channels[cnum].origin == origin &&
-        (comp[comp_sound] || channels[cnum].is_pickup == is_pickup))
+        (default_comp[comp_sound] || channels[cnum].is_pickup == is_pickup))
```

Comment:
EE replaces the check for `is_pickup` with one for `singularity` but doesn't apply any other checks.